# Prefer NumPy Array

## Why

NumPy **arrays** provide vectorized operations and better memory efficiency.

## Bad practice

```python
arr = [1, 2, 3]
```

A little explanation about the bad practice

## Good Practice

```python
import numpy as np
arr = np.array([1, 2, 3])
```

A little explanation about the good practice

## Compared to [A particular Language]

How it works
For instance, in C an array contains only éléments of the same datatype, ....

```c
int numbers[5] = {1, 2, 3, 4, 5};
```

## Performance

Talk about the antipattern in terms of performance
